modules/effects/arrow-effects/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -77,6 +77,8 @@ sealed class IO<out A> : IOOf<A> {
         }
       }
 
+    val never: IO<Nothing> = async { _, _ -> Unit }
```

Comment:
This should come from the Async instance.

Reply:
What is the advantage? It ends up calling this same code after three indirections and `async` is also public and in scope.

Reply:
I think it'd be better if the IOAsyncInstance called this directly instead of `async` in the type class so that there is less indirection

Reply:
I don't have a strong preference, just pointing out that codegen should bring that constructor in anyway. IIRC it was there at some point.